lib/oga/xml/document.rb

```diff
@@ -57,6 +57,12 @@ def root_node
         self
       end
 
+      # Returns the top level element in the document.
+
```

Comment:
Hmm, I hadn't looked into the `HTML` module, but had assumed there was a parallel class tree there specialized for HTML-specific behavior.

To me it seems inevitable that some modality will be introduced (outside of just the parser) for the differences between XML and HTML.  It'd be simple enough for now to just set a field on the document from the parser, and then have methods like this do different things based on that field.  I get that that's not particularly elegant (but then, we're talking about parsing HTML), but leaving out behavior that everyone who's ever used an XML parser will expect because of HTML's quirks seems broken.